File: 2021/Day7/Median of two sorted arrays/median_of two_sorted arrays.cpp
class Solution
{
public:
    double findMedianSortedArrays(vector<int> &nums1, vector<int> &nums2)
    {
        double med = 0.0;
        for (int val : nums2)
        {
            nums1.push_back(val);
        }
        sort(nums1.begin(), nums1.end());
        int n = nums1.size();
        med = (n % 2 == 0) ? (double)(nums1[n / 2] + nums1[(n / 2) - 1]) / 2 : nums1[n / 2];

        return med;
    }
};

File: 2021/Day7/Median of two sorted arrays/median_of_two_sorted_arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        nums3 = sorted(nums1 + nums2)
        x = len(nums3)
        if(x % 2 == 0):
            x = x//2
            a = nums3[x]
            x = x-1
            a = float(a + nums3[x])/2.0
            return a
        else:
            x = x//2
            return (float(nums3[x]))
